File: recursion_3.py
#!/usr/bin/python3

def base10ToString(number):
    binary_string = []

    def str_maker(number):
        if number < 2:
            binary_string.append(number)
            return
        else:
            str_maker(int(number / 2))
            str_maker(number % 2)

    str_maker(number)
    return binary_string


# print(base10ToString(13))
arr = base10ToString(232)
for i in arr:
    print("{}".format(i), end="")
print()

File: recursion_3.js
#!/usr/bin/node

base10ToString = (n) => {
    // Convert a base 10 number into
    // a base 2 number
    var binaryString = "";

    base10ToStringHelper = (n) => {
        // Recursive function
        if (n < 2){
            binaryString += n;
            return;
        }
        else {
            base10ToStringHelper(Math.floor(n / 2));
            base10ToStringHelper(n % 2);
        }
    }
    base10ToStringHelper(n);

    return binaryString;
}

console.log(base10ToString(13))
